internal/network/data_sender.go

```diff
@@ -67,9 +67,24 @@ func NewDataSender(unitID uuid.UUID, addr common.Address, logger zerolog.Logger,
 	// Setup mqtt connection. Does not connect
 	isSecureConn := conf.Mqtt.Broker.TLS.Enabled
 
+	// Set the logger for the MQTT client. Uncomment to enable debug logging
+	//mqtt.DEBUG = log.New(os.Stdout, "[DEBUG] ", 0)
+
 	// Create MQTT client options
 	opts := mqtt.NewClientOptions()
 	opts.AddBroker(conf.Mqtt.Broker.Host + ":" + strconv.Itoa(conf.Mqtt.Broker.Port))
+	// this will allow to publish messages even if connection is not established and saving messages in file store.
+	// with waitWithTimeout on connect failure, we still can publish messages which would be stored in fileStore
+	opts.SetConnectRetry(true)
+	// messages buffering in file store, default is "in memory" store for Qos1 and 2
+	opts.SetStore(mqtt.NewFileStore("/opt/autopi/store"))
+	// indicates that the client should store the messages in the file store after shutdown and pickup messages upon start up from the disk
+	// if we shut down the edge-network and start again, all buffered messages will be deleted on startup unless we set SetCleanSession to false
+	opts.SetCleanSession(false)
+	// ConnectRetryInterval is the time to wait between reconnection attempts. Default is  30 sec
+	opts.SetConnectRetryInterval(time.Second * 10)
+	// If we are using SetCleanSession=false, we need to specify client-id
+	opts.SetClientID(addr.String())
```

Comment:
As we need to have a client ID, we need to come up with a unique client ID for every device. Is the addr a good candidate?